plugin.video.hrmagplayer/libplayer/player.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+import xbmc
+import xbmcgui
+import xbmcplugin
+import xbmcaddon
+import sys
+from libplayer.channel import *
+from libplayer.utils import *
+
+# Get the plugin url in plugin:// notation.
+__url__ = sys.argv[0]
+# Get the plugin handle as an integer number.
+__handle__ = int(sys.argv[1])
+
+__context__ = None
+
+class HrMagPlayer(xbmc.Player):
```

Comment:
Plugins must not use or extend the `xbmc.Player` class. The player class is to be used in scripts where you create your own interface via skinning. Plugins are simply vfs directory listings and should use `xbmcplugin.setResolvedUrl()` instead